scripts/lint

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -e
+
+echo
+echo " -- CHECKING TYPES WITH PYRIGHT --"
+echo
+
+scripts/pyright pystac tests
```

Comment:
I'd recommend running this and other lints on setup.py as well. Another approach would be to gather all Python files which are version controlled (`git ls-files '*.py'`) and list each of them, like this:

```
mapfile -d '' python_files < <(git ls-files -z '*.py)
scripts/pyright "${python_files[@]}"
```

Reply:
> I'd recommend running this and the other linters on setup.py as well.

Are you suggesting making the linting part of the install process for this package or that we allow people to run the tests using `setup.py`? 

If the first option, it seems like that would required adding the linting requirements (`mypy`, `flake8`, etc.) to the build dependencies and would make the install process take longer. I'm not sure I see the advantage given that we are running this as part of the CI for every PR and every time we publish to PyPi. 

If the second, the `test` command [was deprecated](https://github.com/pypa/setuptools/issues/1684) from setuptools, so I don't think we want to try to support that, unless there is another supported way of running those from setup.py.

Reply:
Neither, I was suggesting that setup.py should also be linted. :)

Reply:
Hah, yes, good suggestion!

Reply:
> Another approach would be to gather all Python files which are version controlled (`git ls-files '*.py'`) and list each of them

Okay, this makes more sense to me now in the context of the other part of the comment. I'll experiment with this and see how it works.

Reply:
[Related](https://github.com/stac-utils/pystac/issues/390)

Reply:
Since this is going to change based on #401 anyway, I just added `setup.py` to the list of files being linted (and to the autoformatting in `scripts/format`).